index.js

```diff
@@ -0,0 +1,311 @@
+        case 'Medium priority':
+            li.style.backgroundColor = '#a3cd3b';
+            break;
+        case 'High priority':
+            li.style.backgroundColor = '#ff4943';
+            break;
+    }
+}
+
+function createNewTask() {
+    editTaskButton.style.display = "none";
+    addTaskButton.style.display = "inline";
+}
+
+
+function addTask() {
+    const taskCount = taskField.getElementsByTagName("li").length
+    let task = {
+        title: '',
+        text: '',
+        priority: '',
+        time: '',
+        date: ''
+
+    }
+    task.title = title.value;
+    task.text = text.value;
+    task.priority = selectPriority() + ' priority';
+    task.time = setTaskTime();
+    task.date = new Date();
+    toDo.push(task);
+    createTaskField(task, taskField);
+    toDoCount++
+    document.getElementById('toDo-count').innerHTML = ' (' + toDoCount + ')';
+    saveData();
+}
+
+function moveToComplete() {
+    let currentTask = this.closest('.list-group-item');
+    index = Array.from(currentTask.parentNode.children).indexOf(currentTask);
+    taskField.removeChild(currentTask);
+    completed.push(toDo[index]);
+    createTaskField(toDo[index], completeField)
+    toDo.splice(index, 1);
+    toDoCount--
+    document.getElementById('toDo-count').innerHTML = ' (' + toDoCount + ')';
+    completedToDoCount++
+    document.getElementById('completed-count').innerHTML = ' (' + completedToDoCount + ')';
+    saveData();
+}
+
+function editTaskData() {
+    addTaskButton.style.display = "none";
+    editTaskButton.style.display = "inline";
+    let currentTask = this.closest('.list-group-item');
+    index = Array.from(currentTask.parentNode.children).indexOf(currentTask);
+    title.value = toDo[index].title;
+    text.value = toDo[index].text;
+    switch (toDo[index].priority) {
+        case 'Low priority':
+            document.getElementById('Low').checked = "true";
+            break;
+        case 'Medium priority':
+            document.getElementById('Medium').checked = "true";
+            break;
+        case 'High priority':
+            document.getElementById('High').checked = "true";
+            break;
+        default:
+            document.getElementById('Low').checked = "true";
+            break;
+    }
+
+}
+
+function saveEditChanges() {
+    let li = document.querySelectorAll('.list-group-item')[index];
+    let task = {
+        title: '',
+        text: '',
+        priority: '',
+        time: ''
+    }
+    task.title = title.value;
+    task.text = text.value;
+    task.priority = selectPriority() + ' priority';
+    task.time = setTaskTime();
+    toDo.splice(index, 1, task);
+    let newTitle = taskField.children[index].getElementsByTagName("h5")[0];
+    let newText = taskField.children[index].getElementsByTagName("p")[0];
+    let newPriority = taskField.children[index].getElementsByTagName("small")[0];
+    let newTime = taskField.children[index].getElementsByTagName("small")[1];

```

Comment:
don't use this strange search with index in the end, we have more accurate methods for this